mlrun/kfpops.py

```diff
@@ -596,3 +609,138 @@ def get_default_reg():
+    workflow = run["pipeline_runtime"].get("workflow_manifest", None)
+    if not workflow:
+        return None
+    workflow = json.loads(workflow)
+
+    templates = {}
+    for template in workflow["spec"]["templates"]:
+        project = project or get_in(
+            template, ["metadata", "annotations", project_annotation], ""
+        )
+        name = template["name"]
+        templates[name] = {
+            "run_type": get_in(
+                template, ["metadata", "annotations", run_annotation], ""
+            ),
+            "function": get_in(
+                template, ["metadata", "annotations", function_annotation], ""
+            ),
+        }
+
+    nodes = workflow["status"].get("nodes", {})
+    dag = {}
+    # name_map = {k: v["displayName"] for k, v in nodes.items()}
+    for node in nodes.values():
+        name = node["displayName"]
+        record = {
+            k: node[k] for k in ["phase", "startedAt", "finishedAt", "type", "id"]
+        }
+        record["parent"] = node.get("boundaryID", "")
+        record["name"] = name
+        record["children"] = node.get("children", [])
+        if name in templates:
+            record["function"] = templates[name].get("function")
+            record["run_type"] = templates[name].get("run_type")
+        dag[node["id"]] = record
+
+    return dag, project
+
+
+def format_summary_from_kfp_run(kfp_run, project=None, session=None):
+    override_project = project if project and project != "*" else None
+    dag, project = generate_kfp_dag_and_resolve_project(kfp_run, override_project)
+    run_id = get_in(kfp_run, "run.id")
+
+    # enrich DAG with mlrun run info
+    if session:
+        runs = mlrun.api.utils.singletons.db.get_db().list_runs(
+            session, project=project, labels=f"workflow={run_id}"
+        )
+    else:
+        runs = get_run_db().list_runs(project=project, labels=f"workflow={run_id}")
+
+    for run in runs:
+        step = get_in(run, ["metadata", "labels", "mlrun/runner-pod"])
+        if step and step in dag:
+            dag[step]["run_uid"] = get_in(run, "metadata.uid")
+            dag[step]["kind"] = get_in(run, "metadata.labels.kind")
+            error = get_in(run, "status.error")
+            if error:
+                dag[step]["error"] = error
+
+    short_run = {"graph": dag}
+    short_run["run"] = {
+        k: str(v)
+        for k, v in kfp_run["run"].items()
+        if k
+        in [
+            "id",
+            "name",
+            "status",
+            "error",
+            "created_at",
+            "scheduled_at",
+            "finished_at",
+            "description",
+        ]
+    }
+    short_run["run"]["project"] = project
+    return short_run
+
+
+def show_kfp_run(run, clear_output=False):
+    phase_to_color = {"failed": "red", "succeeded": "green", "skipped": "white"}
+    runtype_to_shape = {"run": "ellipse", "build": "box", "deploy": "box3d"}
```

Comment:
I would suggest adding an enum for the `pipeline_run_type` too and use it here and in the `add_annotations` calls as well